### PR TITLE
Appointment

### PR DESCRIPTION
**Pull request by bacancy-priyank**

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -237,6 +247,9 @@ PLATFORMS
 
 DEPENDENCIES
   bootsnap (>= 1.4.2)
+  bootstrap-datepicker-rails
+  bootstrap-sass
+  bootstrap-timepicker-rails
   byebug
   capybara (>= 2.15)
   city-state
```
Comment (pinkal-bacancy) [inline Gemfile.lock:255]: Add gemfile.lock in gitignore. Don't push this file on git

File: app/controllers/medico_controller.rb [hunk 2]
```diff
@@ -12,10 +12,76 @@ def home; end
...
     @hospitals = Hospital.where(admin_id: current_user.id)
   end
+
   def hospital
-    @hospital = Hospital.find(params[:id])
+    @doctors = Doctor.where(hospital_id: Hospital.find(params[:id]))
+  end
+  
+  def doctor
+    @doctor = Doctor.find_by(id: params[:id])
```
Comment (pinkal-bacancy) [inline app/controllers/medico_controller.rb:21]: Why you have used find_by instead of find?

File: app/controllers/medico_controller.rb [hunk 3]
```diff
@@ -12,10 +12,76 @@ def home; end
...
+    @appointment = @doctor.appointments.build
```
Comment (pinkal-bacancy) [inline app/controllers/medico_controller.rb:22]: You should use @appointments here.

File: app/controllers/medico_controller.rb [hunk 4]
```diff
@@ -12,10 +12,76 @@ def home; end
...
+    @patient_id = current_user.id
```
Comment (pinkal-bacancy) [inline app/controllers/medico_controller.rb:23]: Is the @patient_id in use in the view file? If not then don't use @.

Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb:23]: yes

File: app/controllers/medico_controller.rb (hunk 3 above)
Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb:22]: It is for booking one appointment

File: Gemfile.lock (hunk 1 above)
Comment (bacancy-priyank) [reply Gemfile.lock:255]: it is already in .gitignore

File: app/controllers/medico_controller.rb [hunk 5]
```diff
@@ -12,10 +12,76 @@ def home; end
...
+    @availability= @doctor.availability
```
Comment (pinkal-bacancy) [inline app/controllers/medico_controller.rb:24]: Is the @availability in use in the view file? If not then don't use @.

Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb:24]: yes, it is used for ajax

File: app/controllers/medico_controller.rb [hunk 6]
```diff
@@ -12,10 +12,76 @@ def home; end
...
+    @disabled_days = []
+    (0..6).each do |value|
+      if !value.between?(@availability.start_day, @availability.end_day)
```
Comment (pinkal-bacancy) [inline app/controllers/medico_controller.rb:27]: Make a scope or method in model to find out available day.

Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb:27]: okay

File: app/controllers/medico_controller.rb (hunk 2 above)
Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb]: resolved

File: app/views/layouts/partials/_user_footer.html.erb [hunk 7]
```diff
@@ -1,79 +1,39 @@
...
+            <p></p>
           </div>
       </div>
     </div>
-  </div>
-<!-- //bootstrap-modal-pop-up --> 
-  <!-- js -->
-<%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
\ No newline at end of file
+  </div>
```
[the remark this answers is not in the document]
Comment (pinkal-bacancy) [reply app/views/layouts/partials/_user_footer.html.erb:39]: Add one line at the end if the file

File: app/controllers/medico_controller.rb [hunk 8]
```diff
@@ -12,10 +12,76 @@ def home; end
...
+  
+  def appointments
+    @appointments = Appointment.where(patient_id: current_user.id)
+    @date = Date.today
   end
+  
+  def cancel_appointment
+    appointment = Appointment.find(params[:id])
+    if appointment.destroy
+      flash[:notice]="Appointment is deleted successfully"
```
[the remark this answers is not in the document]
Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb:58]: done

File: app/controllers/medico_controller.rb [hunk 9]
```diff
@@ -12,10 +12,76 @@ def home; end
...
+      redirect_to appointments_medico_index_path
+    else
+      flash[:notice]="Apoointment cannot be deleted"
```
[the remark this answers is not in the document]
Comment (bacancy-priyank) [reply app/controllers/medico_controller.rb:61]: done

File: app/views/hospitals/index.html.erb [hunk 10]
```diff
@@ -1,36 +1,30 @@
...
-    <%end%>
-  </table>
-  <%= link_to "New Hospital", new_hospital_path,class:"btn btn-primary"%>
-    <%# <div class="digg_pagination">
-  <div class="page_info">
-    <%= page_entries_info @bookings %>
-    <%# </div>   %>
-    <%# <%= will_paginate @bookings, renderer: BootstrapPagination::Rails %>
-    <%# </div>  %>
+<% if  !@hospitals.empty? %>
```
[the remark this answers is not in the document]
Comment (bacancy-priyank) [reply app/views/hospitals/index.html.erb:1]: object @hospitals lidho che to chalse?>

File: app/models/doctor.rb [hunk 11]
```diff
@@ -1,10 +1,19 @@
...
+  belongs_to :user, class_name: "User", optional: true
   has_many :patients, through: :appointments, foreign_key: 'patient_id'
   has_many :appointments
   has_one :availability
   accepts_nested_attributes_for :availability, allow_destroy: true, reject_if: :all_blank
+
+  def self.disabled_days(availability)
+    disabled_days = []
+    (0..6).each do |value|
+      if !value.between?(availability.start_day, availability.end_day)
```
Comment (pinkal-bacancy) [inline app/models/doctor.rb:14]: Use unless instead of if !

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -9,24 +9,23 @@
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= stylesheet_pack_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <%# <%= render 'layouts/partials/user_head' %> 
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-timepicker/1.10.0/jquery.timepicker.css">
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-timepicker/1.10.0/jquery.timepicker.js"></script>
     <!-- Custom fonts for this template-->
-    <link href="/assets/all.min.css" rel="stylesheet" type="text/css">
+    <%# <link href="/assets/all.min.css" rel="stylesheet" type="text/css">
```
Comment (pinkal-bacancy) [inline app/views/layouts/application.html.erb:16]: Remove this code if not needed.

Comment (bacancy-priyank) [reply app/views/layouts/application.html.erb:16]: it is creating issues but it is prachi's code so i need to discuss

**Pull request merged**